### PR TITLE
Add support for user public events.

### PR DESCRIPTION
**Pull request by NinjaTrappeur**
Implement improvement #273

This PR implements a get method as well as its associated tests for this endpoint: https://developer.github.com/v3/activity/events/#list-public-events-performed-by-a-user

Since the get repository events endpoint (https://developer.github.com/v3/activity/events/#list-repository-events) was quite similar and was missing integration tests, I took advantage of this PR to write some.

I tried to respect as much as I could your coding style regarding tests, hope I managed to do that right.

Please note that I am a beginner regarding haskell, do not hesitate to let me know if I can improve something. I will not be offended and glad to fix that!

Have a nice day!

### CODE REVIEW

File: spec/GitHub/EventsSpec.hs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+spec = do
+  describe "repositoryEventsR" $ do
+    it "returns non empty list of events" $ shouldSucceed $
+      GitHub.repositoryEventsR "phadej" "github" 1
+  describe "userEventsR" $ do
+    it "returns non empty list of events" $ shouldSucceed $ GitHub.userEventsR "phadej" 1 
+  where shouldSucceed f = withAuth $ \auth -> do
+          cs <- GitHub.executeRequest auth $ f
+          cs `shouldSatisfy` isRight
+          length (fromRightS cs) `shouldSatisfy` (> 1)
```
Comment (phadej) [inline spec/GitHub/EventsSpec.hs]: use

```
import Prelude ()
import Prelude.Compat
```

to get correct (polymorphic) `length`.

File: spec/GitHub/EventsSpec.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    Just token -> action (OAuth $ fromString token)
+
+spec :: Spec
+spec = do
+  describe "repositoryEventsR" $ do
+    it "returns non empty list of events" $ shouldSucceed $
+      GitHub.repositoryEventsR "phadej" "github" 1
+  describe "userEventsR" $ do
+    it "returns non empty list of events" $ shouldSucceed $ GitHub.userEventsR "phadej" 1 
+  where shouldSucceed f = withAuth $ \auth -> do
```
Comment (phadej) [inline spec/GitHub/EventsSpec.hs]: Note to self: refactor this when merged (quicker to do myself, than try to explain what I want).

**Comment by NinjaTrappeur** [comment]
Thanks for the quick feedback.

Alright, I'm going to get that pass through travis. 

I'll squash, force push and let you know when I'm done.

**Comment by NinjaTrappeur** [comment]
Alright, it should be ok now. I'll just let travis run.

I was watching travis's logs and noticed it lacks a github API hence integration tests are skipped:

>GitHub.Activity
 watchersForR
    works
     # PENDING: no GITHUB_TOKEN
  myStarredR
    works
     # PENDING: no GITHUB_TOKEN
GitHub.Commits
  commitsFor
    works
     # PENDING: no GITHUB_TOKEN
    limits the response
     # PENDING: no GITHUB_TOKEN

I do not know if you are aware about that…

I ran these tests locally, they all pass using the lts-7 Haskell.

**Comment by phadej** [comment]
yeah, GITHUB_TOKEN is private and not set for non-member builds (otherwise you could expose them, by modifying `.travis.yml`); that's not an issue.

I'll look at this tomorrow

**Comment by NinjaTrappeur** [comment]
Hey @phadej, friendly bump.

**Pull request merged**